lib/util.js

```diff
@@ -1030,13 +1030,16 @@ function error(...args) {
 }
 
 function _errnoException(err, syscall, original) {
-  var name = errname(err);
+  if (typeof err !== 'number' || err >= 0 || !Number.isSafeInteger(err)) {
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'err',
+                               'negative number');
+  }
+  const name = errname(err);
   var message = `${syscall} ${name}`;
   if (original)
     message += ` ${original}`;
-  var e = new Error(message);
-  e.code = name;
-  e.errno = name;
+  const e = new Error(message);
```

Comment:
IMHO a new type something like `errors.UVError` or `errors.PlatformError` would be nice, but then we'd need to expose it somehow.